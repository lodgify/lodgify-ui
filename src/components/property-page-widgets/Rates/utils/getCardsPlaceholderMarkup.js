import React, { Fragment } from 'react';
import { Card } from 'semantic-ui-react';

import { TextPlaceholder } from 'elements/TextPlaceholder';
import { Divider } from 'elements/Divider';

/**
 * @return {Object}
 */
export const getCardsPlaceholderMarkup = () => (
  <Fragment>
    {['first', 'second', 'third'].map(item => (
      <Card fluid key={item}>
        <Card.Content>
          <Divider />
          <TextPlaceholder length="short" />
          <Divider />
          <TextPlaceholder length="medium" />
          <TextPlaceholder length="medium" />
          <TextPlaceholder length="medium" />
          <Divider size="large" />
          <TextPlaceholder />
          <Divider />
          <TextPlaceholder />
          <Divider />
          <TextPlaceholder />
          <Divider />
        </Card.Content>
      </Card>
    ))}
  </Fragment>
);
