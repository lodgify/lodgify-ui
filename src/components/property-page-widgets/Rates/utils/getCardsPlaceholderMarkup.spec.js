import { mount } from 'enzyme';

import { getCardsPlaceholderMarkup } from './getCardsPlaceholderMarkup';

describe('getCardsPlaceholderMarkup', () => {
  it('should render the right structure', () => {
    const actual = mount(getCardsPlaceholderMarkup());

    expect(actual).toMatchSnapshot();
  });
});
