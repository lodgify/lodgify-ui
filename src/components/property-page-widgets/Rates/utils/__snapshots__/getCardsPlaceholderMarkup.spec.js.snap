// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCardsPlaceholderMarkup should render the right structure 1`] = `
Array [
  <Card
    fluid={true}
    key="first"
  >
    <div
      className="ui fluid card"
      onClick={[Function]}
    >
      <CardContent>
        <div
          className="content"
        >
          <Divider
            className=""
            hasLine={false}
            size="medium"
          >
            <Divider
              className=""
              hidden={true}
              section={false}
            >
              <div
                className="ui hidden divider"
              />
            </Divider>
          </Divider>
          <TextPlaceholder
            isFluid={true}
            length="short"
          >
            <div
              className="placeholder text short fluid"
            />
          </TextPlaceholder>
          <Divider
            className=""
            hasLine={false}
            size="medium"
          >
            <Divider
              className=""
              hidden={true}
              section={false}
            >
              <div
                className="ui hidden divider"
              />
            </Divider>
          </Divider>
          <TextPlaceholder
            isFluid={true}
            length="medium"
          >
            <div
              className="placeholder text medium fluid"
            />
          </TextPlaceholder>
          <TextPlaceholder
            isFluid={true}
            length="medium"
          >
            <div
              className="placeholder text medium fluid"
            />
          </TextPlaceholder>
          <TextPlaceholder
            isFluid={true}
            length="medium"
          >
            <div
              className="placeholder text medium fluid"
            />
          </TextPlaceholder>
          <Divider
            className=""
            hasLine={false}
            size="huge"
          >
            <Divider
              className="is-size-huge"
              hidden={true}
              section={false}
            >
              <div
                className="ui hidden divider is-size-huge"
              />
            </Divider>
          </Divider>
          <TextPlaceholder
            isFluid={true}
            length="full"
          >
            <div
              className="placeholder text full fluid"
            />
          </TextPlaceholder>
          <Divider
            className=""
            hasLine={false}
            size="medium"
          >
            <Divider
              className=""
              hidden={true}
              section={false}
            >
              <div
                className="ui hidden divider"
              />
            </Divider>
          </Divider>
          <TextPlaceholder
            isFluid={true}
            length="full"
          >
            <div
              className="placeholder text full fluid"
            />
          </TextPlaceholder>
          <Divider
            className=""
            hasLine={false}
            size="medium"
          >
            <Divider
              className=""
              hidden={true}
              section={false}
            >
              <div
                className="ui hidden divider"
              />
            </Divider>
          </Divider>
          <TextPlaceholder
            isFluid={true}
            length="full"
          >
            <div
              className="placeholder text full fluid"
            />
          </TextPlaceholder>
          <Divider
            className=""
            hasLine={false}
            size="medium"
          >
            <Divider
              className=""
              hidden={true}
              section={false}
            >
              <div
                className="ui hidden divider"
              />
            </Divider>
          </Divider>
        </div>
      </CardContent>
    </div>
  </Card>,
  <Card
    fluid={true}
    key="second"
  >
    <div
      className="ui fluid card"
      onClick={[Function]}
    >
      <CardContent>
        <div
          className="content"
        >
          <Divider
            className=""
            hasLine={false}
            size="medium"
          >
            <Divider
              className=""
              hidden={true}
              section={false}
            >
              <div
                className="ui hidden divider"
              />
            </Divider>
          </Divider>
          <TextPlaceholder
            isFluid={true}
            length="short"
          >
            <div
              className="placeholder text short fluid"
            />
          </TextPlaceholder>
          <Divider
            className=""
            hasLine={false}
            size="medium"
          >
            <Divider
              className=""
              hidden={true}
              section={false}
            >
              <div
                className="ui hidden divider"
              />
            </Divider>
          </Divider>
          <TextPlaceholder
            isFluid={true}
            length="medium"
          >
            <div
              className="placeholder text medium fluid"
            />
          </TextPlaceholder>
          <TextPlaceholder
            isFluid={true}
            length="medium"
          >
            <div
              className="placeholder text medium fluid"
            />
          </TextPlaceholder>
          <TextPlaceholder
            isFluid={true}
            length="medium"
          >
            <div
              className="placeholder text medium fluid"
            />
          </TextPlaceholder>
          <Divider
            className=""
            hasLine={false}
            size="huge"
          >
            <Divider
              className="is-size-huge"
              hidden={true}
              section={false}
            >
              <div
                className="ui hidden divider is-size-huge"
              />
            </Divider>
          </Divider>
          <TextPlaceholder
            isFluid={true}
            length="full"
          >
            <div
              className="placeholder text full fluid"
            />
          </TextPlaceholder>
          <Divider
            className=""
            hasLine={false}
            size="medium"
          >
            <Divider
              className=""
              hidden={true}
              section={false}
            >
              <div
                className="ui hidden divider"
              />
            </Divider>
          </Divider>
          <TextPlaceholder
            isFluid={true}
            length="full"
          >
            <div
              className="placeholder text full fluid"
            />
          </TextPlaceholder>
          <Divider
            className=""
            hasLine={false}
            size="medium"
          >
            <Divider
              className=""
              hidden={true}
              section={false}
            >
              <div
                className="ui hidden divider"
              />
            </Divider>
          </Divider>
          <TextPlaceholder
            isFluid={true}
            length="full"
          >
            <div
              className="placeholder text full fluid"
            />
          </TextPlaceholder>
          <Divider
            className=""
            hasLine={false}
            size="medium"
          >
            <Divider
              className=""
              hidden={true}
              section={false}
            >
              <div
                className="ui hidden divider"
              />
            </Divider>
          </Divider>
        </div>
      </CardContent>
    </div>
  </Card>,
  <Card
    fluid={true}
    key="third"
  >
    <div
      className="ui fluid card"
      onClick={[Function]}
    >
      <CardContent>
        <div
          className="content"
        >
          <Divider
            className=""
            hasLine={false}
            size="medium"
          >
            <Divider
              className=""
              hidden={true}
              section={false}
            >
              <div
                className="ui hidden divider"
              />
            </Divider>
          </Divider>
          <TextPlaceholder
            isFluid={true}
            length="short"
          >
            <div
              className="placeholder text short fluid"
            />
          </TextPlaceholder>
          <Divider
            className=""
            hasLine={false}
            size="medium"
          >
            <Divider
              className=""
              hidden={true}
              section={false}
            >
              <div
                className="ui hidden divider"
              />
            </Divider>
          </Divider>
          <TextPlaceholder
            isFluid={true}
            length="medium"
          >
            <div
              className="placeholder text medium fluid"
            />
          </TextPlaceholder>
          <TextPlaceholder
            isFluid={true}
            length="medium"
          >
            <div
              className="placeholder text medium fluid"
            />
          </TextPlaceholder>
          <TextPlaceholder
            isFluid={true}
            length="medium"
          >
            <div
              className="placeholder text medium fluid"
            />
          </TextPlaceholder>
          <Divider
            className=""
            hasLine={false}
            size="huge"
          >
            <Divider
              className="is-size-huge"
              hidden={true}
              section={false}
            >
              <div
                className="ui hidden divider is-size-huge"
              />
            </Divider>
          </Divider>
          <TextPlaceholder
            isFluid={true}
            length="full"
          >
            <div
              className="placeholder text full fluid"
            />
          </TextPlaceholder>
          <Divider
            className=""
            hasLine={false}
            size="medium"
          >
            <Divider
              className=""
              hidden={true}
              section={false}
            >
              <div
                className="ui hidden divider"
              />
            </Divider>
          </Divider>
          <TextPlaceholder
            isFluid={true}
            length="full"
          >
            <div
              className="placeholder text full fluid"
            />
          </TextPlaceholder>
          <Divider
            className=""
            hasLine={false}
            size="medium"
          >
            <Divider
              className=""
              hidden={true}
              section={false}
            >
              <div
                className="ui hidden divider"
              />
            </Divider>
          </Divider>
          <TextPlaceholder
            isFluid={true}
            length="full"
          >
            <div
              className="placeholder text full fluid"
            />
          </TextPlaceholder>
          <Divider
            className=""
            hasLine={false}
            size="medium"
          >
            <Divider
              className=""
              hidden={true}
              section={false}
            >
              <div
                className="ui hidden divider"
              />
            </Divider>
          </Divider>
        </div>
      </CardContent>
    </div>
  </Card>,
]
`;
